### PR TITLE
[On-chain] feat: add `proof_window_close_offset_blocks` as `shared` module parameter

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Adds `poktroll.shared.Params#proof_window_close_offset_blocks` to the `shared` module parameters.

## Issue

- #516

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: x/shared/keeper/msg_server_update_param_test.go [hunk 1]
```diff
@@ -124,3 +124,31 @@ func TestMsgUpdateParam_UpdateProofWindowOpenOffsetBlocks(t *testing.T) {
...
+	msgSrv := keeper.NewMsgServerImpl(k)
+
+	// Set the parameters to their default values
+	defaultParams := sharedtypes.DefaultParams()
+	require.NoError(t, k.SetParams(ctx, defaultParams))
+
+	// Ensure the default values are different from the new values we want to set
+	require.NotEqual(t, uint64(expectedProofWindowCloseOffestBlocks), defaultParams.ProofWindowCloseOffsetBlocks)
+
+	// Update the claim window close offset blocks param
```
Comment (bryanchriswhite) [inline x/shared/keeper/msg_server_update_param_test.go:141]:
```suggestion
	// Update the proof window close offset blocks param
```

**Review by red-0ne — APPROVED** [review]
LGTM

**Review by red-0ne — COMMENTED** [review]
Isn't it missing params_test and a make target to update it?

**Review by Olshansk — CHANGES_REQUESTED** [review]
Things we seem to have elsewhere:
1. Makefie target
2. Accompanying .json file
3. Unit test for failing update

**Review by Olshansk — APPROVED** [review]

**Pull request merged**